examples/usage.rs

```diff
@@ -0,0 +1,58 @@
+#![feature(try_blocks)]
+// Overrides the std library's `Result` type.
+
+use std::fs::File;
+use std::io;
+
+enum MyError {
+    Unlucky,
+    Io(io::Error),
+    TooSmall(u64),
+}
+
+impl From<io::Error> for MyError {
+    fn from(e: io::Error) -> Self {
+        Self::Io(e)
+    }
+}
+
+fn file_size(path: &str) -> propagate::Result<u64, MyError> {
+    try {
+        // The `?` operator coerces `std::result::Result<_, io::Error>` into `Result<_, MyError>`.
+        let size = File::open(path)?.metadata()?.len();
+
+        if size < 1024 {
+            Err(MyError::TooSmall(size))?
+        }
+
+        size
+    }
+}
```

Comment:
I don't love the extra indentation from the try block but I think it's worth it to avoid having to shadow `Ok`

Reply:
Yeah I'm not sure folks at my work will be in love with that either, so it's somewhat important to me that the code still looks decent even if the user doesn't use `try` at all.